### PR TITLE
python3 + flake8

### PR DESCRIPTION
**Pull request by gesape**
Hi Marius, läuft unter python3 problemlos (und produziert die gleichen Ergebnisse wie unter python2).
An einigen Stellen, bei denen ich mir nicht ganz sicher bin, habe ich Kommentare eingefügt...

### CODE REVIEW

File: autogain/autogain.py [hunk 1]
```diff
@@ -203,22 +219,24 @@ def __init__(self, event, stations):
     def finish(self, reference_nsl, fband, taper):
         for tr in self.traces:
             tr_backup = tr.copy()
-            tr_backup.set_location('B' )
-            tr.ydata -= tr.get_ydata().mean()
+            tr_backup.set_location('B')
+            tr.ydata -= num.int32(tr.get_ydata().mean())  # otherwise dtype error!
```
Comment (HerrMuellerluedenscheid) [inline autogain/autogain.py]: What if the trace's ydata is of type num.float? Can you try if this works?

    tr.ydata -= tr.get_ydata().mean().view(tr.ydata.dtype)

This should (I guess) cast the mean always to the dtype of tr.ydata.

File: autogain/autogain.py [hunk 2]
```diff
@@ -285,35 +306,45 @@ def process(self, fband, taper):
...
-                tr = self.data_pile.chopper(tmin=event.time+arrival - window_min,
-                                            tmax=event.time+arrival + window_max,
+                tr = self.data_pile.chopper(tmin=event.time+arrival -
+                                            window_min,
+                                            tmax=event.time+arrival +
+                                            window_max,
                                             trace_selector=selector)
-                _tr = tr.next()
+                try:
+                    _tr = next(tr)    # does not work without try and except?!
```
Comment (HerrMuellerluedenscheid) [inline autogain/autogain.py]: Hmm.... was I doing the same thing as in line 342 here? Probably I accidentally skipped every second trace :-(

Comment (HerrMuellerluedenscheid) [reply autogain/autogain.py]: Ah, no that was just to make sure that each chunk of data containes just one trace. Puh, ugly way of doing this though...

File: autogain/autogain.py (hunk 1 above)
Comment (gesape) [reply autogain/autogain.py]: You're right, I will try that.

File: autogain/autogain.py (hunk 2 above)
Comment (gesape) [reply autogain/autogain.py]: Nothing is skipped, but I guess it really is not a beautiful way to check for having only one trace. It works fine, though, but without the added try and except I had problems when there was no next trace anymore.

Comment (HerrMuellerluedenscheid) [reply autogain/autogain.py]: Ah, sorry, thought that was my creation :-)

File: autogain/autogain.py (hunk 1 above)
Comment (gesape) [reply autogain/autogain.py]: That does not work, but this, although it does not look that beautiful


dtype = type(tr.ydata[0])
tr.ydata -= dtype(tr.get_ydata().mean())


tr.ydata -= tr.get_ydata().mean().view(tr.ydata.dtype) does not work, either.

**Comment by HerrMuellerluedenscheid** [comment]
Nice!

**Pull request merged**